File: web/src/api/notebook.ts
import service from '@/utils/request';

// 获取 Notebook 列表
export const getNotebookList = (params: any) => {
    return service({
        url: '/api/v1/notebook/list',
        method: 'get',
        params
    });
};

// 创建 Notebook
export const createNotebook = (data: any) => {
    return service({
        url: '/api/v1/notebook/add',
        method: 'post',
        data
    });
};

// 删除 Notebook
export const deleteNotebook = (data: any) => {
    return service({
        url: '/api/v1/notebook/delete',
        method: 'post',
        data
    });
};

// 停止 Notebook
export const stopNotebook = (data: any) => {
    return service({
        url: '/api/v1/notebook/stop',
        method: 'post',
        data
    });
};

// 启动 Notebook
export const startNotebook = (data: any) => {
    return service({
        url: '/api/v1/notebook/start',
        method: 'post',
        data
    });
};

// 获取 Notebook 详情
export const getNotebookDetail = (params: any) => {
    return service({
        url: '/api/v1/notebook/get',
        method: 'get',
        params
    });
};

// 获取 Notebook Pod 列表
export const getNotebookPods = (params: any) => {
    return service({
        url: '/api/v1/notebook/pod/list',
        method: 'get',
        params
    });
};

// 获取 Notebook 日志
export const getNotebookLogs = (params: any) => {
    return service({
        url: '/api/v1/notebook/log/list',
        method: 'get',
        params
    });
};

// 更新 Notebook
export const updateNotebook = (data: any) => {
    return service({
        url: '/api/v1/notebook/update',
        method: 'post',
        data
    });
};

// Notebook 终端 (WebSocket)
export const getNotebookTerminal = (params: any) => {
    return service({
        url: '/api/v1/notebook/terminal',
        method: 'get',
        params
    });
};

// 终端及身份认证
export const getNotebookAuth = (params: any) => {
    return service({
        url: '/api/v1/notebook/auth',
        method: 'get',
        params
    });
};

// Notebook 日志流 (WebSocket)
export const getNotebookLogStream = (params: any) => {
    return service({
        url: '/api/v1/notebook/log/stream',
        method: 'get',
        params
    });
};

File: server/router/notebook/notebook.go
package notebook

import (
	"gin-vue-admin/middleware"

	"github.com/gin-gonic/gin"
)

type NotebookRouter struct{}

func (s *NotebookRouter) InitNotebookApiRouter(PrivateGroup *gin.RouterGroup, PublicGroup *gin.RouterGroup) {
	notebookRouter := PrivateGroup.Group("notebook").Use(middleware.OperationRecord())
	notebookWithNotRecorderRouter := PrivateGroup.Group("notebook")
	notebookPublicRouter := PublicGroup.Group("notebook")
	{
		notebookRouter.POST("add", notebookApi.AddNoteBook)       // 创建容器实例
		notebookRouter.POST("delete", notebookApi.DeleteNoteBook) // 删除容器实例
		notebookRouter.POST("update", notebookApi.UpdateNoteBook) // 更新容器实例
		notebookRouter.POST("stop", notebookApi.StopNotebook)     // 停止容器实例
		notebookRouter.POST("start", notebookApi.StartNotebook)   // 启动容器实例
	}
	{
		// WebSocket 路由（无法携带 Header，通过 query token 鉴权）
		notebookPublicRouter.GET("terminal", notebookApi.HandleTerminal)     // Web Terminal (WebSocket)
		notebookPublicRouter.GET("log/stream", notebookApi.HandleStreamLogs) // 日志流 (WebSocket)
	}
	{
		notebookWithNotRecorderRouter.GET("list", notebookApi.GetNotebookList)     // 获取容器实例
		notebookWithNotRecorderRouter.GET("get", notebookApi.GetNotebookDetail)    // 获取容器实例详情
		notebookWithNotRecorderRouter.GET("pod/list", notebookApi.GetNotebookPods) // 获取容器实例 Pod 列表
		notebookWithNotRecorderRouter.GET("log/list", notebookApi.GetNotebookLogs) // 获取容器实例日志
	}
}
